sdks/python/apache_beam/dataframe/frames.py

```diff
@@ -1120,6 +1222,15 @@ def agg(self, fn):
             requires_partition_by=partitionings.Index(),
             preserves_partition_by=partitionings.Singleton()))
 
+  aggregate = agg
+
+  first = last = head = tail = frame_base.not_implemented_method(
+      'order sensitive')
+
+  __len__ = frame_base.wont_implement_method('non-deferred')
```

Comment:
Should we consider implementing this and `groups` for categorical grouping keys?

Reply:
Good thought. Dropped a TODO for now.